### ISSUE TITLE
use recommended labels for PMEM-CSI deployments

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#850]
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

This replaces our current usage of "app" as label, which is ambiguous: does it name the component independently of the CSI driver name or does it encode the driver name? We can avoid that with:
```
        app.kubernetes.io/name: pmem-csi-controller # fixed
        app.kubernetes.io/part-of: pmem-csi # fixed
        app.kubernetes.io/component: controller # fixed
        app.kubernetes.io/instance: pmem-csi.intel.com #CSI driver name, varies
```

This is slightly redundant, but that's intentional because it provides more flexibility for filtering.

**Comment by pohly** [comment]
I am making that change for the PMEM-CSI driver because I need it for supporting the scheduler extensions with more than one driver.

I am leaving deploy/kustomize/operator/operator.yaml unchanged for now.

**Comment by pohly** [comment]
I am leaving out `app.kubernetes.io/version` for now. Whether we want to add that can be treated as a separate issue.

**Comment by pohly** [comment]
Solved as part of PR #838
